File: src/ProjectR/content/scripts/afflictions/Passives/AggresiveRegeneration.lua
function GetName()
  return "Aggressive Regeneration"
end

function GetType()
  return Passive
end

function GetHookPoints()
  return { TurnTriggered }
end

function OnAttachment(char)
end

function OnRemoval(char)
end

function Trigger(hookPoint, char)
  if hookPoint ~= TurnTriggered then
    return
  end

  local amount = (char:GetMaxHP() - char:GetCurrentHP()) * 0.25
  char:Heal(amount)
end

File: src/ProjectR.Scripting/content/scripts/afflictions/Passives/AggresiveRegeneration.cs
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting.Afflictions
{
    public class AggresiveRegeneration : Affliction
    {
        public override string Name { get { return "Aggressive Regeneration"; } }

        public override AfflictionType Type { get { return AfflictionType.Passive; } }

        protected override HookPoint[] HookPoints
        {
            get
            {
                return new[]
                {
                    HookPoint.TurnTriggered, 
                };
            }
        }

        protected override void OnTurnTriggered(ICharacter character)
        {
            var amount = (character.Stats.GetTotalStat(BaseStat.HP) - character.CurrentHP) * .25;
            character.Heal(amount);
        }
    }
}